### PR TITLE
dashboard: make menus configurable

### PR DESCRIPTION
**Pull request by martinobersteiner**
:heart: Thank you for your contribution!

### Description

Currently, submenus on the user dashboard [are hard-coded](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/users_ui/views/ui.py#L70), making their configuration hard.
This PR moves them to a config-variable, which allows to:
- more easily add further submenus (by way of configuration)
- make use of submenus' many options by configuring the corresponding kwarg, e.g. `visible_when`-kwarg to tune visibility

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Comment by Samk13** [comment]
Thanks for the changes. Makes sense to me👍

File: invenio_app_rdm/config.py [hunk 1]
```diff
@@ -689,6 +689,25 @@ def files_rest_permission_factory(obj, action):
     "requests": "/me/requests",
 }
 
+APP_RDM_USER_DASHBOARD_SUBMENUS = {
```
Comment (kpsherva) [inline invenio_app_rdm/config.py:692]: I have some concerns about expanding the config variable matrix for small UI improvements, I would rather try to find more sustainable solution, but I need some time to propose something, please reach me back to discuss

**Review by lnielsen — APPROVED** [review]

**Comment by kpsherva** [comment]
finally: agreed on having more flexible and generic approach

**Review by slint — APPROVED** [review]
LGTM! Thanks a lot for the changes.

**Pull request merged**